pkg/trace/obfuscate/sql.go

```diff
@@ -151,22 +152,58 @@ func (f *groupingFilter) Reset() {
 // Process the given SQL or No-SQL string so that the resulting one is properly altered. This
 // function is generic and the behavior changes according to chosen tokenFilter implementations.
 // The process calls all filters inside the []tokenFilter.
-func obfuscateSQLString(in string) (string, error) {
-	tokenizer := NewSQLTokenizer(in)
+func (o *Obfuscator) obfuscateSQLString(in string) (string, error) {
+	ignoreEscapes := o.ShouldIgnoreEscapes()
+	tokenizer := NewSQLTokenizer(in, ignoreEscapes)
 	filters := []tokenFilter{&discardFilter{}, &replaceFilter{}, &groupingFilter{}}
+
+	out, err := attemptObfuscation(tokenizer, filters)
+	if err != nil {
+		// If the tokenizer failed, but saw an escape character in the process
+		// retry with opposite setting.
+		if tokenizer.HasSeenEscape() {
+			tokenizer = NewSQLTokenizer(in, !ignoreEscapes)
+			var err2 error
+			out, err2 = attemptObfuscation(tokenizer, filters)
```

Comment:
This error handling is a bit tricky. Can we simplify it?
```go
out, err := attemptObfuscation(tokenizer, filters)
if err != nil {
    return "", err
}
// our assumption was correct; change the default behaviour
o.SetLiteralEscapes(!literalEscapes)
return out, err
```

Reply:
I think we want to return the original error if the second attempt fails.

Reply:
You're right. I had a feeling I missed something here.